sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -191,10 +192,13 @@ export class AppConfigurationClient {
     options: AddConfigurationSettingOptions = {}
   ): Promise<AddConfigurationSettingResponse> {
     return this._trace("addConfigurationSetting", options, async (newOptions) => {
+      const internalConfigurationSetting = serializeAsConfigurationSettingParam(
+        configurationSetting
+      );
       const originalResponse = await this.client.putKeyValue(configurationSetting.key, {
         ifNoneMatch: "*",
         label: configurationSetting.label,
-        entity: configurationSetting,
+        entity: internalConfigurationSetting,
```

Comment:
If you wanted to have a distinct name you could call it KeyValue (that's the actual entity name used by AppConfig in the rest layer)

Reply:
Calling it keyValue now